hydromt/models/model.py

```diff
@@ -313,127 +300,58 @@ def update(
             try to write to a file that's already opened. The output will be written
             to a temporary file in case the original file cannot be written to.
         """
-        opt = opt or {}
-        opt = self._check_get_opt(opt)
+        steps = steps or {}
+        validate_steps(self, steps)
+
+        # check if region.create is in the steps, and remove it.
+        if steps.pop("region.create", None) is not None:
+            self.logger.warning(
+                "region.create can only be called when building a model."
+            )
 
         # read current model
         if not self.root.is_writing_mode():
             if model_out is None:
                 raise ValueError(
-                    '"model_out" directory required when updating in "read-only" mode'
+                    '"model_out" directory required when updating in "read-only" mode.'
                 )
             self.read()
-            if forceful_overwrite:
-                self.root.set(model_out, mode="w+")
-            else:
-                self.root.set(model_out, mode="w")
+            mode = "w+" if forceful_overwrite else "w"
+            self.root.set(model_out, mode=mode)
 
         # check if model has a region
         if self.region.data is None:
             raise ValueError("Model region not found, setup model using `build` first.")
 
-        # remove CLI_ARGS method from options and throw warning
-        method = self._CLI_ARGS["region"]
-        if method in opt:
-            opt.pop(method)  # remove from opt
-            self.logger.warning(f'"{method}" can only be called when building a model.')
-
-        # loop over other methods from config file
-        for method in opt:
-            # if any write_* functions are present in opt, skip the final self.write()
-            if "write" in method:
-                write = False
-            kwargs = {} if opt[method] is None else opt[method]
-            self._run_log_method(method, **kwargs)
-
-        # write
-        if write:
+        # loop over methods from config file
+        for step in steps:
+            self.logger.info(f"update: {step}")
+            kwargs = steps[step] or {}
+            # Call the methods.
+            rgetattr(self, step)(kwargs)
```

Comment:
Same note for the log messages when running the method

Reply:
added this back.